lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/restart/DFSEvaluatorLogAppendReaderWriter.java

```diff
@@ -65,6 +66,11 @@ public synchronized void writeToEvaluatorLog(final String formattedEntry) throws
     }
   }
 
+  @Override
+  public Iterable<String> readFromEvaluatorLog() throws IOException {
```

Comment:
That said, we can return some sort of `CloseableIterable` class, though the logic there might be a bit hazy, i.e. do we `close` upon no more next elements? Or do we wait for the user to call `close`?